### PR TITLE
Add code coverage for ListBoxAccessibleObject and ListBoxItemAccessib…

### PR DESCRIPTION
**Pull request by Syareel-Sukeri**
Related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit test for ListBoxAccessibleObject to test its property: State 
- Add unit test for ListBoxAccessibleObject to test its method: GetChild, GetFocused, GetSelected, HitTest, GetChildCount
- Add unit test for ListBoxItemAccessibleObject to test its properties: Bounds, DefaultAction, Role
- Add unit test for ListBoxItemAccessibleObject to test its method: DoDefaultAction, Select

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11724)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 1]
```diff
@@ -159,4 +159,96 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
         Assert.Equal(expected, !result.IsEmpty && (bool)result);
         Assert.False(listBox.IsHandleCreated);
     }
+
+    [WinFormsFact]
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:163]: Please enable nullability in this portion of the file at least.
```
#nullable enable

<code>
#nullable disable
````

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good, I added only a minor comment.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 2]
```diff
@@ -159,4 +159,100 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
...
+
+        form.Show();
+
+        listBox.Height = 200;
+        listBox.Width = 100;
+
+        var boundsAfterHandleCreation = itemAccessibleObject.Bounds;
+    }
+
+    # nullable disable
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:183]:
```suggestion
    #nullable disable
```

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:183]: Is it possible to enable nullability in all newly added tests?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 3]
```diff
@@ -159,4 +159,100 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
         Assert.Equal(expected, !result.IsEmpty && (bool)result);
         Assert.False(listBox.IsHandleCreated);
     }
+
+    #nullable enable
+
+    [WinFormsFact]
+    public void ListBoxItemAccessibleObject_Bounds_BeforeAndAfterHandleCreation()
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:166]: Are we missing some validation codes in this test case?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 4]
```diff
@@ -246,4 +247,166 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
 
         return accessibilityObject;
     }
+
+    #nullable enable
+
+    [WinFormsTheory]
+    [InlineData(false, AccessibleStates.Focusable)]
+    [InlineData(true, AccessibleStates.Focused | AccessibleStates.Focusable)]
+    public void ListBoxAccessibleObject_State_ShouldBeExpected(bool focusControl, AccessibleStates expectedState)
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:256]: Maybe we can update case like this:
```
[WinFormsFact]
public void ListBoxAccessibleObject_State_ShouldBeExpected()
{
    using Form form = new();
    using ListBox listBox = new();
    using TextBox textBox = new();
    form.Controls.Add(listBox);
    form.Controls.Add(textBox);

    form.Show();
    listBox.CreateControl();

    textBox.Focus();
    listBox.Focused.Should().BeFalse();
    listBox.AccessibilityObject.State.Should().Be(AccessibleStates.Focusable);

    listBox.Focus();
    listBox.Focused.Should().BeTrue();
    listBox.AccessibilityObject.State.Should().Be(AccessibleStates.Focused | AccessibleStates.Focusable);
}
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 5]
```diff
@@ -246,4 +247,166 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
...
+        accessibleObject.GetSelected().Should().BeNull();
+
+        listBox.SelectedIndex = 0;
+        accessibleObject.GetSelected().Should().Be(accessibleObject.GetChild(0));
+
+        listBox.SelectedIndex = 1;
+        accessibleObject.GetSelected().Should().Be(accessibleObject.GetChild(1));
+
+        listBox.ClearSelected();
+        accessibleObject.GetSelected().Should().BeNull();
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:297]: We can add a verification that ListBox Handle is not created to the end of the test.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 6]
```diff
@@ -246,4 +247,166 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
...
+    [InlineData(2, 2)]
+    public void GetChildCount_ReturnsExpected(int numberOfItemsToAdd, int expectedCount)
+    {
+        using ListBox listBox = new();
+        for (int i = 0; i < numberOfItemsToAdd; i++)
+        {
+            listBox.Items.Add($"Item {i + 1}");
+        }
+
+        var accessibleObject = new ListBox.ListBoxAccessibleObject(listBox);
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:311]: We can simplify this ListBox.ListBoxAccessibleObject(listBox);

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 7]
```diff
@@ -246,4 +247,166 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
...
+
+        int childCount = accessibleObject.GetChildCount();
+
+        childCount.Should().Be(expectedCount);
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:315]: Simplify this as accessibleObject.GetChildCount().Should().Be(expectedCount);

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 8]
```diff
@@ -246,4 +247,166 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
...
+        using ListBox listBox = new();
+        listBox.Items.AddRange(new[] { "Item 1", "Item 2" });
+        listBox.CreateControl();
+        listBox.SelectedIndex = 1; // Focus the second item
+        listBox.Focus();
+
+        var accessibleObject = listBox.AccessibilityObject;
+        var focusedObject = accessibleObject.GetFocused();
+
+        focusedObject.Should().BeEquivalentTo(accessibleObject.GetChild(1));
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:330]: We can add a verification that ListBox Handle is not created to the end of the test.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 9]
```diff
@@ -246,4 +247,166 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
...
+                listBox.SelectedIndices.Add(2); // Select an additional item for multiple selection scenarios
+            }
+        }
+
+        if (clearSelection)
+        {
+            listBox.ClearSelected();
+        }
+
+        var accessibleObject = new ListBox.ListBoxAccessibleObject(listBox);
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:356]: We can simplify this ListBox.ListBoxAccessibleObject(listBox);

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 10]
```diff
@@ -246,4 +247,166 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
...
+    [WinFormsFact]
+    public void TestHitTest_ListBoxNotCreated_ReturnsNull()
+    {
+        using Form form = new();
+        using ListBox listBox = new() { Parent = form, Items = { "Item 1", "Item 2" } };
+        Point testPoint = new Point(10, 10);
+
+        var result = listBox.AccessibilityObject.HitTest(testPoint.X, testPoint.Y);
+
+        result.Should().BeNull();
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:378]: We can add a verification that ListBox Handle is not created to the end of the test.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 11]
```diff
@@ -159,4 +159,100 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
...
+    }
+
+    [WinFormsFact]
+    public void ListBoxItemAccessibleObject_DefaultAction_VariesByContext()
+    {
+        using Form formDoubleClick = new();
+        using ListBox listBoxDoubleClick = new() { Items = { "Item 1" }, Parent = formDoubleClick };
+        formDoubleClick.Show();
+
+        var itemAccessibleObjectDoubleClick = listBoxDoubleClick.AccessibilityObject.GetChild(0) as ListBox.ListBoxItemAccessibleObject;
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:190]: We don't need to do type conversions.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 12]
```diff
@@ -159,4 +159,100 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
...
+    }
+
+    [WinFormsFact]
+    public void ListBoxItemAccessibleObject_VerifyProperties()
+    {
+        using ListBox listBox = new();
+        listBox.Items.Add("Item 1");
+
+        var accessibleObject = listBox.AccessibilityObject;
+        var itemAccessibleObject = accessibleObject.GetChild(0) as ListBox.ListBoxItemAccessibleObject;
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:206]: We don't need to do type conversions.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 13]
```diff
@@ -159,4 +159,100 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
...
+    }
+
+    [WinFormsFact]
+    public void TestDoDefaultAction_HandleCreatedAndNotCreated()
+    {
+        using Form form = new();
+        using ListBox listBox = new() { Parent = form };
+        listBox.Items.Add("Test Item");
+
+        var itemAccessibleObject = listBox.AccessibilityObject.GetChild(0) as ListBox.ListBoxItemAccessibleObject;
```
Comment (SimonZhao888) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:225]: We don't need to do type conversions.

**Review by Tanya-Solyanik — COMMENTED** [review]
@SimonZhao888  - please review to make you all your comments are resolved correctly.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 14]
```diff
@@ -187,13 +194,14 @@ public void ListBoxItemAccessibleObject_DefaultAction_VariesByContext()
...
+        var itemAccessibleObjectDoubleClick = (ListBox.ListBoxItemAccessibleObject?)listBoxDoubleClick.AccessibilityObject.GetChild(0);
+        itemAccessibleObjectDoubleClick.Should().NotBeNull();
+        itemAccessibleObjectDoubleClick!.DefaultAction.Should().Be("Double Click");
 
         using ListBox listBoxNullAction = new ListBox { Items = { "Item 2" } };
-        var itemAccessibleObjectNullAction = listBoxNullAction.AccessibilityObject.GetChild(0) as ListBox.ListBoxItemAccessibleObject;
-
-        itemAccessibleObjectNullAction?.DefaultAction.Should().BeNull();
+        var itemAccessibleObjectNullAction = (ListBox.ListBoxItemAccessibleObject?)listBoxNullAction.AccessibilityObject.GetChild(0);
+        itemAccessibleObjectNullAction.Should().NotBeNull();
```
[the remark this answers is not in the document]
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:203]: I mean that you can satisfy nullability guarantees using casts like this:

    [WinFormsFact]
    public void ListBoxItemAccessibleObject_DefaultAction_VariesByContext()
    {
        using Form formDoubleClick = new();
        using ListBox listBoxDoubleClick = new() { Items = { "Item 1" }, Parent = formDoubleClick };
        formDoubleClick.Show();

        var itemAccessibleObjectDoubleClick = listBoxDoubleClick.AccessibilityObject.GetChild(0).Should().BeOfType<ListBox.ListBoxItemAccessibleObject>().Which;
        itemAccessibleObjectDoubleClick.DefaultAction.Should().Be("Double Click");

        using ListBox listBoxNullAction = new ListBox { Items = { "Item 2" } };
        var itemAccessibleObjectNullAction = listBoxNullAction.AccessibilityObject.GetChild(0).Should().BeOfType<ListBox.ListBoxItemAccessibleObject>().Which;
        itemAccessibleObjectDoubleClick.DefaultAction.Should().BeNull();
    }

This pattern can be applies to other tests in this change as well.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 15]
```diff
@@ -159,4 +159,105 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
...
+    [WinFormsTheory]
+    [InlineData(AccessibleSelection.AddSelection)]
+    [InlineData(AccessibleSelection.RemoveSelection)]
+    [InlineData(AccessibleSelection.TakeFocus)]
+    [InlineData((AccessibleSelection)int.MaxValue)]
+    public void Select_WithVariousFlags_ShouldNotThrow(AccessibleSelection flags)
+    {
+        using ListBox listBox = new();
+        ItemArray.Entry itemEntry = new ItemArray.Entry("Test Item");
+        ListBox.ListBoxAccessibleObject listBoxAccessibleObject = new ListBox.ListBoxAccessibleObject(listBox);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:253]:
```suggestion
        ListBox.ListBoxAccessibleObject listBoxAccessibleObject = new(listBox);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs [hunk 16]
```diff
@@ -159,4 +159,105 @@ public void ListBoxItemAccessibleObject_GetPropertyValue_Pattern_ReturnsExpected
...
+        ListBox.ListBoxItemAccessibleObject listBoxItemAccessibleObject = new ListBox.ListBoxItemAccessibleObject(listBox, itemEntry, listBoxAccessibleObject);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBox.ListBoxItemAccessibleObjectTests.cs:254]:
```suggestion
        ListBox.ListBoxItemAccessibleObject listBoxItemAccessibleObject = new(listBox, itemEntry, listBoxAccessibleObject);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs [hunk 17]
```diff
@@ -246,4 +247,162 @@ private ListBoxAccessibleObject InitListBoxItemsAccessibleObjects(ListBox listBo
...
+        listBox.IsHandleCreated.Should().BeFalse();
+    }
+
+    [WinFormsFact]
+    public void TestHitTest_PointInsideListBoxBoundsButOutsideItems_ReturnsSelf()
+    {
+        using Form form = new();
+        using ListBox listBox = new() { Parent = form, Items = { "Item 1", "Item 2" } };
+        listBox.CreateControl();
+        form.Show();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:383]: @lonitra  - we are moving all tests that show UI to the System.Windows.Forms.UI.IntegrationTests, right?

Comment (lonitra) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs:383]: Yes, any UI test should be in UI.IntegrationTests

Comment (Syareel-Sukeri) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/ListBoxAccessibleObjectTests.cs]: I have relocated all the UI-related tests from this pull request to the `System.Windows.Forms.UI.IntegrationTests` project. Please review the changes at your convenience.

**Review by SimonZhao888 — DISMISSED** [review]

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListBoxAccesibleObjectTests.cs [hunk 18]
```diff
@@ -0,0 +1,59 @@
+﻿using System.Drawing;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListBoxAccesibleObjectTests.cs:1]: Please add a copyright header
```
﻿// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
```

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**